jujugui/static/gui/src/app/components/status/status.js

```diff
@@ -45,142 +40,40 @@ class Status extends React.Component {
-        return true;
-      }
+  _setTrafficLight() {
+    const db = this.props.db;
+    let statuses = [];
+    db.services.toArray().forEach(application => {
+      const app = application.getAttrs();
+      statuses.push(app.status.current);
     });
-    return status;
-  }
-
-  /**
-    Return an element class name suitable for the given value.
-    @param {String} prefix The class prefix.
-    @param {String} value The provided value.
-    @returns {String} The class name ('ok', 'error' or '').
-  */
-  _getStatusClass(prefix, value) {
-    if (!value) {
-      // If there is no value then ignore it. This might be the case when an
-      // entity's state property only has a value for pending/error states.
-      return '';
-    }
-    if (!Array.isArray(value)) {
-      value = [value];
-    }
-    const normalised = value.map(val => this._normaliseStatus(val));
-    return prefix + this._getHighestStatus(normalised);
-  }
-
-  /**
-    Normalise the status value.
-    @param status {String} The raw value.
-    @returns {String} The normalised status ('ok', 'error' or 'pending').
-  */
-  _normaliseStatus(value) {
-    let status = this.STATUS_OK;
-    switch(value) {
-      case 'active':
-      case 'idle':
-      case 'started':
-      case 'waiting':
-        status = this.STATUS_OK;
-        break;
-      case 'blocked':
-      case 'down':
-      case 'error':
-        status = this.STATUS_ERROR;
-        break;
-      case 'pending':
-      case 'installing':
-      case 'executing':
-      case 'allocating':
-      case 'maintenance':
-        status = this.STATUS_PENDING;
-        break;
-    }
-    return status;
-  }
-
-  /**
-    Generate a status for display.
-    @param status {String} The status to display.
-    @returns {Object} The status markup.
-  */
-  _generateStatusDisplay(status) {
-    // If the status provided from a model property it might have no value.
-    if (!status) {
-      return null;
-    }
-    return (
-      <span className={this._getStatusClass('status-view__status--', status)}>
-        {status}
-      </span>);
-  }
-
-  /**
-    Sort by the key attribute.
-    @param {Object} a The first value.
-    @param {Object} b The second value.
-    @returns {Array} The sorted array.
-  */
-  _byKey(a, b) {
-    if (a.key < b.key) {
-      return -1;
-    }
-    if (a.key > b.key) {
-      return 1;
-    }
-    return 0;
-  }
-
-  /**
-    Filter a row by the status.
-    @param row {Object} The row values.
-    @returns {Boolean} Whether the row matches the status.
-  */
-  _filterByStatus(row) {
-    if (!this.state.statusFilter) {
-      return true;
+    db.units.toArray().forEach(unit => {
+      statuses.push(utils.getHighestStatus([unit.agentStatus, unit.workloadStatus]));
+    });
+    db.machines.toArray().map(machine => {
+      statuses.push(machine.agent_state);
+    });
+    const highest = utils.getHighestStatus(statuses);
+    if (highest && (this.state.highestStatus !== highest)) {
```

Comment:
I'm not sure this check is strictly necessary, should it not just set the highest status and then let the component figure out what to do with that value?